model/src/main/java/ru/training/karaf/repo/BookRepoImpl.java

```diff
@@ -0,0 +1,176 @@
+
+public class BookRepoImpl implements BookRepo {
+    private JpaTemplate template;
+
+    public BookRepoImpl(JpaTemplate template) {
+        this.template = template;
+    }
+
+    @Override
+    public List<Book> getAll() {
+        return template.txExpr(em -> em.createNamedQuery(BookDO.GET_ALL, BookDO.class).getResultList())
+                .stream()
+                .map(BookImpl::new)
+                .collect(Collectors.toList());
+
+    }
+
+    @Override
+    public void create(String name, String usr, int price, int red, int green, int blue, int alpha, Collection<String> authors) {
+        BookDO bookToCreate = new BookDO();
+        ColorImpl color = new ColorImpl(red, green, blue, alpha);
+        bookToCreate.setName(name);
+        bookToCreate.setColor(color);
+
+
+        /*BookPriceRepoImpl bookPriceRepo = new BookPriceRepoImpl(template);
+        long l = bookPriceRepo.create(price);
+        bookToCreate.getPrice().add(template.txExpr(em -> em.find(BookPriceDO.class, l)));*/
+
+        bookToCreate.setAuthors(template.txExpr(em -> em.createNamedQuery(AuthorDo.GET_WHERE_IN, AuthorDo.class).setParameter("authors", authors)
+                .getResultList()));
+
+        UserDO userDO = template.txExpr(em -> em.createNamedQuery(UserDO.GET_BY_LOGIN, UserDO.class).setParameter("login", usr).getSingleResult());
+        bookToCreate.setUsr(userDO);
+
+        Long id = template.txExpr(em -> em.merge(bookToCreate)).getId();
+        BookPriceRepoImpl bookPriceRepo = new BookPriceRepoImpl(template);
+        bookPriceRepo.create(price, id);
+
+    }
+
+
+    @Override
+    public Optional<Book> get(long id) {
+        return template.txExpr(em -> getById(id, em)).map(BookImpl::new);
+    }
+
+    @Override
+    public void delete(String name) {
+        template.tx(em -> em.createNamedQuery(BookDO.DELETE_BY_NAME, BookDO.class).setParameter("name", name).executeUpdate());
+    }
+
+    @Override
+    public List<Book> getByUsr(String usr) {
+        return template.txExpr(em -> em.createNamedQuery(BookDO.GET_ALL_BY_USR, BookDO.class).setParameter("usr", usr).getResultList())
+                .stream()
+                .map(BookImpl::new)
+                .collect(Collectors.toList());
+    }
+
+    private Optional<BookDO> getById(long id, EntityManager em) {
+        try {
+            return Optional.of(em.createNamedQuery(BookDO.GET_BY_ID, BookDO.class).setParameter("id", id)
+                    .getSingleResult());
+        } catch (NoResultException e) {
+            return Optional.empty();
+        }
+    }
+
+    @Override
+    public void changePrice(String name, String author, int price) {
+        Optional<BookDO> bookDO = template.txExpr(em -> getByNameAndAuthor(em, name, author));
+        if (bookDO.isPresent()) {
+            BookDO book = bookDO.get();
+            BookPriceRepoImpl bookPriceRepo = new BookPriceRepoImpl(template);
+            bookPriceRepo.create(price, book.getId());
```

Comment:
Here you already have BookDO class but you will fetch it from db one more time in this method. Let's try to keep the number of DB requests as low as possible.

However, sometimes it's ok to make additional db request if it needed to keep code clean and simple. I think it's not the case here though.